packages/frontend/src/hooks/useSpaces.ts

```diff
@@ -35,13 +42,13 @@ export const useSpaceSummaries = (
         {
             select: (data) =>
                 // only get spaces that the user has direct access to
-                !includePrivateSpaces
-                    ? data.filter(
+                includePrivateSpaces
+                    ? data
+                    : data.filter(
                           (space) =>
-                              !space.isPrivate ||
-                              (!!user && space.access.includes(user.userUuid)),
-                      )
-                    : data,
+                              !!user &&
+                              hasDirectAccessToSpace(space, user.userUuid),
+                      ),
```

Comment:
You can move the user check to the top. If the user is not defined, then it would just return an empty list, right? Although, it does make me think if that edge case is even possible :think: ... If not, you could just throw an error.

Reply:
The reason for putting it there is to narrow down the ts type I think.